codec_packetizers/h265/h265_depacketizer.c

```diff
@@ -0,0 +1,505 @@
+static H265Result_t H265Depacketizer_ProcessFragmentationUnit( H265DepacketizerContext_t * pCtx,
+                                                               H265Nalu_t * pNalu )
+{
+    uint8_t * pCurPacketData;
+    uint8_t fuHeader = 0;
+    uint8_t firstByte, secondByte;
+    size_t curPacketLength;
+    size_t curNaluDataIndex = 0;
+    H265Result_t result = H265_RESULT_OK;
+
+/* Process fragments until we reach the end or run out of packets */
+    while( ( pCtx->packetCount > 0 ) &&
+           ( ( fuHeader & FU_HEADER_E_BIT_MASK ) == 0 ) )
+    {
+        pCurPacketData = pCtx->pPacketsArray[pCtx->tailIndex].pPacketData;
+        curPacketLength = pCtx->pPacketsArray[pCtx->tailIndex].packetDataLength;
+
+/* Check minimum size requirement */
+        if( curPacketLength < TOTAL_FU_HEADER_SIZE )
+        {
+            result = H265_RESULT_MALFORMED_PACKET;
+            break;
+        }
+
+        fuHeader = pCurPacketData[FU_HEADER_OFFSET];
+        firstByte = pCurPacketData[0];
+        secondByte = pCurPacketData[1];
+
+/* Validate packet type */
+        if( ( ( firstByte >> 1 ) & 0x3F ) != FU_PACKET_TYPE )
+        {
+            result = H265_RESULT_MALFORMED_PACKET;
+            break;
+        }
+
+/* Handle start fragment - write NALU header */
+        if( fuHeader & FU_HEADER_S_BIT_MASK )
+        {
+            if( ( curNaluDataIndex + NALU_HEADER_SIZE ) <= pNalu->naluDataLength )
+            {
+                uint8_t fu_type = fuHeader & FU_HEADER_TYPE_MASK;
+                pNalu->pNaluData[curNaluDataIndex] = ( firstByte & 0x81 ) | ( fu_type << 1 );
+                pNalu->pNaluData[curNaluDataIndex + 1] = secondByte;
+            }
+            else
+            {
+                result = H265_RESULT_OUT_OF_MEMORY;
+                break;
+            }
+            curNaluDataIndex += NALU_HEADER_SIZE;
+        }
+
+/* Write NALU payload */
+        size_t payloadLength = curPacketLength - TOTAL_FU_HEADER_SIZE;
+        if( ( curNaluDataIndex + payloadLength ) <= pNalu->naluDataLength )
+        {
+            memcpy( &( pNalu->pNaluData[curNaluDataIndex] ),
+                    &( pCurPacketData[TOTAL_FU_HEADER_SIZE] ),
+                    payloadLength );
+        }
+        else
+        {
+            result = H265_RESULT_OUT_OF_MEMORY;
+            break;
+        }
+        curNaluDataIndex += payloadLength;
+
+/* Move to next packet */
+        pCtx->tailIndex++;
+        pCtx->packetCount--;
+    }
+
+/* Update final NALU length */
+    pNalu->naluDataLength = curNaluDataIndex;
+
+    return result;
+}
+
+
+
+
+/*-------------------------------------------------------------------------------------------------------------*/
+
+static H265Result_t H265Depacketizer_ProcessAggregationPacket( H265DepacketizerContext_t * pCtx,
+                                                               H265Nalu_t * pNalu )
+{
+    H265Result_t result = H265_RESULT_OK;
+    H265Packet_t * pCurrentPacket;
+    size_t minSize;
```

Comment:
Likely minSize is a fixed value. I'd suggest to declare this in this way.
```suggestion
    const size_t minSize = AP_HEADER_SIZE + ( 2 * AP_NALU_LENGTH_FIELD_SIZE + 2 * NALU_HEADER_SIZE );
```